code/game/gamemodes/cult/cult_structures.dm

```diff
@@ -2,29 +2,135 @@
+		return
+	if(cooldowntime)
+		user << "<span class='cultitalic'>The magic here is weak, it will be ready to use again in [getETA()]. </span>"
+		return
+	var/choice = alert(user,"You study the schematics etched into the forge...",,"Eldritch Whetstone","Zealot's Blindfold","Flask of Unholy Water")
+	switch(choice)
+		if("Eldritch Whetstone")
+			var/obj/item/weapon/sharpener/cult/N = new(get_turf(src))
+			user << "<span class='cultitalic'>You kneel before the altar and your faith is rewarded with an [N.name]!</span>"
+		if("Zealot's Blindfold")
+			var/obj/item/clothing/glasses/night/cultblind/N = new(get_turf(src))
+			user << "<span class='cultitalic'>You kneel before the altar and your faith is rewarded with a [N.name]!</span>"
+		if("Flask of Unholy Water")
+			var/obj/item/weapon/reagent_containers/food/drinks/bottle/unholywater/N = new(get_turf(src))
+			user << "<span class='cultitalic'>You kneel before the altar and your faith is rewarded with a [N.name]!</span>"
+	cooldowntime = world.time + 2400
+	spawn(cooldowntime)
+	cooldowntime = 0
+	
 /obj/structure/cult/forge
 	name = "daemon forge"
 	desc = "A forge used in crafting the unholy weapons used by the armies of Nar-Sie"
 	icon_state = "forge"
+	luminosity = 3
+
+/obj/structure/cult/forge/attack_hand(mob/living/user)
+	if(!iscultist(user))
+		user << "<span class='warning'>You don't even begin to understand what these words mean...</span>"
+		return
+	if(cooldowntime)
+		user << "<span class='cultitalic'>The magic here is weak, it will be ready to use again in [getETA()]. </span>"
+		return
+	var/choice = alert(user,"You study the schematics etched into the forge...",,"Shielded Robe","Flagellant's Rags","Nar-Sian Hardsuit")
+	switch(choice)
+		if("Shielded Robe")
+			var/obj/item/clothing/suit/cultrobes/cult_shield/N = new(get_turf(src))
+			user << "<span class='cultitalic'>You work the forge as dark knowledge guides your hands, creating [N]!</span>"
+		if("Flagellant's Rags")
+			var/obj/item/clothing/suit/cultrobes/berserker/N = new(get_turf(src))
+			user << "<span class='cultitalic'>You work the forge as dark knowledge guides your hands, creating [N]!</span>"
+		if("Nar-Sian Hardsuit")
+			new /obj/item/clothing/head/helmet/space/cult(get_turf(src))
+			var /obj/item/clothing/suit/space/cult/N = new(get_turf(src))
+			user << "<span class='cultitalic'>You work the forge as dark knowledge guides your hands, creating [N]!</span>"
+	cooldowntime = world.time + 2400
+	spawn(cooldowntime)
+	cooldowntime = 0
 
 /obj/structure/cult/pylon
 	name = "pylon"
-	desc = "A floating crystal that hums with an unearthly energy"
+	desc = "A floating crystal that slowly heals those faithful to Nar'Sie"
 	icon_state = "pylon"
 	luminosity = 5
+	var/heal_delay = 50
+	var/last_shot = 0
+	var/list/corruption = list()
+
+/obj/structure/cult/pylon/New()
+	SSobj.processing |= src
+	corruption += get_turf(src)
+	for(var/i = 0, i < 4, i++)
+		for(var/turf/T in corruption)
```

Comment:
the first loop should be `for(var/i in 1 to 4)`
the second should be like this:
```
for(var/t in corruption)
    var/turf/T = t
    /* your stuff */
```